app/models/course.rb

```diff
@@ -258,8 +258,16 @@ def remove_site!(site:)
     new? ? site_status.destroy! : site_status.suspend!
   end
 
-  def sites_not_associated_with_course
-    provider.sites - sites
+  def sites=(desired_sites)
+    existing_sites = sites
+
+    to_add = desired_sites - existing_sites
+    to_add.each { |site| add_site!(site: site) }
+
+    to_remove = existing_sites - desired_sites
+    to_remove.each { |site| remove_site!(site: site) }
+
+    sites
```

Comment:
I don't see any tests for this method in the model spec. Can you add some? For example, I'm wondering whether `sites` needs to be reloaded since the methods `add_site!` and `remove_site!` act on the `site_status` objects directly.

Reply:
Is `sites.reload` a thing?..

Reply:
I think so, ya.